src/ObjCRuntime/Blocks.cs

```diff
@@ -83,14 +83,19 @@ public unsafe struct BlockLiteral {
 		[DllImport ("__Internal")]
 		static extern IntPtr xamarin_get_block_descriptor ();
 
+		struct DelegatePairs {
+			public Delegate Trampoline, UserDelegate;
+		}
+
 		//
 		// trampoline must be static, and someone else needs to keep a ref to it
 		//
 		public unsafe void SetupBlock (Delegate trampoline, Delegate userDelegate)
 		{
 			isa = block_class;
 			invoke = Marshal.GetFunctionPointerForDelegate (trampoline);
-			local_handle = (IntPtr) GCHandle.Alloc (userDelegate);
+			var delegates = new DelegatePairs () { Trampoline = trampoline, UserDelegate = userDelegate };
```

Comment:
I _think_ @migueldeicaza used a custom type to get a `struct` not a `class Tuple<X,Y>`.

I wondered if we should have a 

* a new safer/public API (this one) for manual bindings; and 
* an faster/internal one (current one) used for bindings only (can be public with a different name and test to ensure we're not using in manual bindings).

This still needs to fix (cast?) the code in #L160.